consensus/scp/tests/mock_network/mod.rs

```diff
@@ -300,75 +327,84 @@ impl SCPNode {
                                 }
 
-                                // Request to stop thread
-                                SCPNodeTaskMessage::StopTrigger => {
+                                // Stop the thread
+                                SimulatedNodeTaskMessage::StopTrigger => {
                                     break 'main_loop;
                                 }
                             },
                             Err(_) => {
                                 // Yield to other threads when we don't get a new message
-                                // This improves performance significantly.
                                 std::thread::yield_now();
                             }
                         };
 
-                        let incoming_msgs_count = incoming_msgs.len();
-
-                        if !(incoming_msgs_count == 0 || incoming_msgs_count == 1) {
-                            log::error!(logger, "incoming_msgs_count > 1");
-                            // panic
-                            assert!(incoming_msgs_count == 0 || incoming_msgs_count == 1);
-                        }
-
-                        // Process values submitted to our node
-                        if (nominated_values < test_options.max_values_per_slot)
+                        // Nominate pending values submitted to our node
+                        if (nominated_values.len() < max_pending_values_to_nominate)
                             && !pending_values.is_empty()
                         {
-                            let mut vals = pending_values.iter().cloned().collect::<Vec<String>>();
-                            vals.sort();
-                            vals.truncate(test_options.max_values_per_slot - nominated_values);
-                            nominated_values += vals.len();
+                            let mut values: Vec<String> = pending_values.iter().cloned().collect();
+                            values.sort();
+                            values.truncate(max_pending_values_to_nominate);
+
+                            // mc_common::HashSet does not support extend because of our enclave-safe HasherBuilder
+                            let mut selected_values: HashSet<String> = HashSet::default();
+                            for v in values.iter().cloned() {
+                                selected_values.insert(v);
+                            }
 
-                            let outgoing_msg: Option<Msg<String>> = {
-                                thread_local_node
-                                    .lock()
-                                    .expect("lock failed on node nominating value")
-                                    .nominate(current_slot as SlotIndex, BTreeSet::from_iter(vals))
-                                    .expect("node.nominate() failed")
-                            };
+                            let values_to_nominate: HashSet<String> = selected_values
```

Comment:
this is a possible optimization over [byzantine_ledger](https://github.com/mobilecoinofficial/mobilecoin/blob/a323fb78489276178ecdd0ce430cbf2fc24f657b/consensus/service/src/byzantine_ledger.rs#L621rl)